integration/src/test/java/org/mule/test/routing/UntilSuccessfulTestCase.java

```diff
@@ -76,6 +76,43 @@ protected void doTearDown() throws Exception {
     super.doTearDown();
   }
 
+  @Test
+  public void executesOnceWhenNoErrorArisesAndClearsContext() throws Exception {
+    CoreEvent response = flowRunner("happy-path-scope").run();
+    assertThat(getPayloadAsString(response.getMessage()), is("pig"));
+    assertThat(getNumberOfInvocationsFor("insideScope"), is(1));
+    Stack retryContextsStack = ((InternalEvent) response).getInternalParameter("RETRY_CTX");
```

Comment:
asserting the state of the stack would better be unit testing, These integration tests should be more black-box.
Also, ideally, at this point, the stack should have been removed from the context

Reply:
Working on removing it. Will move the tests validating the cleanup of the RetryContext to the scope unit tests.